packages/server/lib/reporter.coffee

```diff
@@ -1,15 +1,32 @@
 _     = require("lodash")
 path  = require("path")
-Mocha = require("mocha")
+## mocha-* is used to allow us to have later versions of mocha specified in devDependencies
+## and prevents accidently upgrading this one
+## TODO: look into upgrading this to version in driver
+Mocha = require("mocha-2.4.5")
+mochaReporters = require("mocha-2.4.5/lib/reporters")
+
 debug = require("debug")("cypress:server:reporter")
 Promise = require("bluebird")
+{ overrideRequire } = require('./overrideRequire')
 
-mochaReporters = require("mocha/lib/reporters")
 
 mochaErrMsgExtractionRe = /^([^:]+): expected/
 
 STATS = "suites tests passes pending failures start end duration".split(" ")
 
+## version 4.1.0 is what we've historically ran our e2e tests against
+## there is no other reason it is specifically 4.1.0
+## TODO: look into upgrading to version in driver
+customReporterMocha = path.dirname(require.resolve('mocha-4.1.0'))
+
+## override calls to `require('mocha*')` when to always resolve with a mocha we control
+## otherwise mocha will be resolved from project's node_modules and might not work with our code
```

Comment:
I don't think this will cause any issues. It should be much easier to support users with custom reporter issues after this change

Reply:
i guess the logic is that it would already be broken if they're using an incompatible version, so this should only improve things for users with broken setups...?

Reply:
yes